doc/rbd/rbd-exclusive-locks.rst

```diff
@@ -27,57 +27,55 @@ Exclusive locking is mostly transparent to the user.
-a hard reboot or power failure of the client node. In that case, the
-exclusive lock is never gracefully released. Thus, when a new process
-starts and attempts to use the device, it needs a way to break the
-previously held exclusive lock.
-
-However, a process (or kernel thread) may also hang, or merely lose
-network connectivity to the Ceph cluster for some amount of time. In
-that case, simply breaking the lock would be potentially catastrophic:
-the hung process or connectivity issue may resolve itself, and the old
-process may then compete with one that has started in the interim,
-accessing RBD data in an uncoordinated and destructive manner.
-
-Thus, in the event that a lock cannot be acquired in the standard
-graceful manner, the overtaking process not only breaks the lock, but
-also blocklists the previous lock holder. This is negotiated between
-the new client process and the Ceph Mon: upon receiving the blocklist
-request,
+Sometimes a client process (or, in case of a krbd client, a client node's
+kernel thread) that previously held an exclusive lock on an image does not
+terminate gracefully, but dies abruptly. This may be because the client process
+received a ``KILL`` or ``ABRT`` signal, or because the client node underwent a
+hard reboot or suffered a power failure. In cases like this, the exclusive lock
+is never gracefully released. This means that any new process that starts and
+attempts to use the device must break the previously held exclusive lock.
+
+However, a process (or kernel thread) may hang or merely lose network
+connectivity to the Ceph cluster for some amount of time. In that case,
+breaking the lock would be potentially catastrophic: the hung process or
+connectivity issue could resolve itself and the original process might then
+compete with one that started in the interim, thus accessing RBD data in an
+uncoordinated and destructive manner.
+
+In the event that a lock cannot be acquired in the standard graceful manner,
+the overtaking process not only breaks the lock but also blocklists the
+previous lock holder. This is negotiated between the new client process and the
+Ceph Mon. Upon receiving the blocklist request:
 
 * the Mon instructs the relevant OSDs to no longer serve requests from
   the old client process;
-* once the associated OSD map update is complete, the Mon grants the
+* after the associated OSD map update is complete, the Mon grants the
   lock to the new client;
-* once the new client has acquired the lock, it can commence writing
+* after the new client has acquired the lock, it can commence writing
   to the image.
 
 Blocklisting is thus a form of storage-level resource `fencing`_.
 
 In order for blocklisting to work, the client must have the ``osd
 blocklist`` capability. This capability is included in the ``profile
-rbd`` capability profile, which should generally be set on all Ceph
+rbd`` capability profile, which should be set generally on all Ceph
```

Comment:
Can you explain this change?  I have always thought the existing phrasing was preferred.

Reply:
I was just trying to keep the modal verb "should" with the verb it was modifying (the passive verb "to be set"). I'm not sure that there's a hard rule in twenty-first century English regarding situations like this, but I'm just trying to keep adverbs out of the middle of multi-word verbs.

If you feel strongly about it, I'm one-hundred percent willing to write it the other way. I think it's better to keep verbs and adverbs separate as much as possible, but it's not a hill I'm going to die on.

Reply:
I'm resolving this thread since you clearly know better than I do -- it's just that the existing wording seemed a lot more common to me.